File: Algorithms/C++/0007/0007.cpp
class Solution {
public:
    int reverse(int x) {
        int rev = 0;
        if (x > 0) {
            while (x != 0) {
                int r = x % 10;
                x /= 10;
                if (rev < INT_MAX / 10 || (rev == INT_MAX / 10 && r <= 7)) {
                    rev = rev * 10 + r;
                } else {
                    rev = 0;
                    break;
                }
            }
        } else {
            while (x != 0) {
                int r = x % 10;
                x /= 10;
                if (rev > INT_MIN / 10 || (rev == INT_MIN / 10 && r >= -8)) {
                    rev = rev * 10 + r;
                } else {
                    rev = 0;
                    break;
                }
            }
        }
        return rev;
    }
};

File: Algorithms/Python/0007/0007.py
class Solution:
    def reverse(self, x: int) -> int:
        rev = 0
        INT_MAX = 2 ** 31 -1
        INT_MIN = -INT_MAX - 1
        negative = False
        if x != INT_MIN:
            if x < 0:
                x, negative = -x, True
            while x != 0:
                r, x = x % 10, x // 10
                if rev < INT_MAX // 10 or (rev == INT_MAX // 10 and r <= 7):
                    rev = rev * 10 + r
                else:
                    rev = 0
                    break
        return rev if not negative else -rev
